controllers/machineset_controller.go

```diff
@@ -156,12 +156,23 @@ func (r *MachineSetReconciler) reconcile(ctx context.Context, machineSet *cluste
 	}
 
 	if cluster != nil && r.shouldAdopt(machineSet) {
+		patch := client.MergeFrom(machineSet.DeepCopy())
 		machineSet.OwnerReferences = util.EnsureOwnerRef(machineSet.OwnerReferences, metav1.OwnerReference{
 			APIVersion: cluster.APIVersion,
 			Kind:       cluster.Kind,
 			Name:       cluster.Name,
 			UID:        cluster.UID,
 		})
+
+		// Patch using a deep copy to avoid overwriting any unexpected Status changes from the returned result
```

Comment:
Why patch twice? The patch is done in a defer call already

Reply:
I had thought so too but there is no `defer Patch` in the `Reconcile` method. This is similar to the behavior in MachineDeployment controller.
Since there was no Patch happening to `updatedMS` I wasn't able to assert on the machine set.
I had a short discussion with Jason [in slack](https://kubernetes.slack.com/archives/C8TSNPY4T/p1570638403003000)

Reply:
Sounds good, seems really an oversight, but we can fix another time

Reply:
@vincepri Sure. I'll split this particular change into another PR.